espflash/src/cli/config.rs

```diff
@@ -94,3 +120,81 @@ impl Config {
             .wrap_err_with(|| format!("Failed to write config to {}", self.save_path.display()))
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use serde::Deserialize;
+
+    #[derive(Debug, Deserialize, Serialize)]
+    struct TestData {
+        #[serde(serialize_with = "parse_u16_hex", deserialize_with = "parse_hex_u16")]
+        value: u16,
+    }
+
+    #[test]
+    fn test_parse_hex_u16() {
+        // Test no padding
+        let input = "aaaa";
+        let result: Result<TestData, _> = toml::from_str(&format!("value = \"{}\"", input));
```

Comment:
probably a matter of taste but I would write something less verbose like
`let result: Result<TestData, _> = toml::from_str(r#"value = "aaaa""#);`

but really a matter of taste